File: backend/src/main/java/ee/rik/domain/service/EventService.java
package ee.rik.domain.service;

import java.util.List;

import ee.rik.domain.Event;
import ee.rik.domain.EventListItem;
import ee.rik.domain.EventParticipant;

public interface EventService {

    List<EventListItem> getAllEvents(Boolean newEvents);

    Event getEvent(Long id);

    Long createEvent(Event event);

    void modifyEvent(Long id, Event event);

    void deleteEvent(Long id);

    List<EventParticipant> listAllParticipants(Long id);

}

File: frontend/src/app/service/event.service.ts
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { EventRepository } from '../repository/event.repository';
import {
  CreateEventRequest,
  Event,
  EventListItem,
  EventParticipant,
  EventParticipantsResponse,
  ListEventsRequest,
  ListEventsResponse,
} from '../generated/rik-backend';

@Injectable({ providedIn: 'root' })
export class EventService {
  public constructor(private readonly repository: EventRepository) {}

  public listNewEvents(): Observable<EventListItem[]> {
    const request: ListEventsRequest = { newEvents: true };
    return this.repository.list(request).pipe(map((response: ListEventsResponse) => response.events));
  }

  public listOldEvents(): Observable<EventListItem[]> {
    const request: ListEventsRequest = { newEvents: false };
    return this.repository.list(request).pipe(map((response: ListEventsResponse) => response.events));
  }

  public getEvent(id: number): Observable<Event> {
    return this.repository.get(id);
  }

  public createEvent(event: Event): Observable<Event> {
    const request: CreateEventRequest = { event };
    return this.repository.create(request);
  }

  public removeEvent(id: number): Observable<void> {
    return this.repository.remove(id);
  }

  public getParticipants(id: number): Observable<EventParticipant[]> {
    return this.repository.getParticipants(id).pipe(map((response: EventParticipantsResponse) => response.participants));
  }
}
